### PR TITLE
spdk: add BdevNvmeSetOptions

### PR DESCRIPTION
**Pull request by derekbit**

### CODE REVIEW

File: pkg/spdk/client/basic.go [hunk 1]
```diff
@@ -560,6 +560,34 @@ func (c *Client) BdevNvmeGetControllers(name string) (controllerInfoList []spdkt
 	return controllerInfoList, json.Unmarshal(cmdOutput, &controllerInfoList)
 }
 
+// BdevNvmeSetOptions sets NVMe driver options.
```
Comment (shuo-wu) [inline pkg/spdk/client/basic.go:563]: Just for confirmation, this API is for attached NVMe controllers, right?

**Review by shuo-wu — APPROVED** [review]
LGTM
It's better to add a cmd for debugging and testing.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@derekbit go merge this if answering @shuo-wu comment.

**Comment by derekbit** [comment]
> LGTM It's better to add a cmd for debugging and testing.

OK. Working on the cmd.

File: pkg/spdk/client/basic.go (hunk 1 above)
Comment (shuo-wu) [reply pkg/spdk/client/basic.go:563]: OK. Plz add this description in this API comment

Comment (derekbit) [reply pkg/spdk/client/basic.go]: Addressed. Thank you.

File: app/cmd/basic/bdev_nvme.go [hunk 2]
```diff
@@ -183,3 +184,51 @@ func bdevNvmeGet(c *cli.Context) error {
 
 	return util.PrintObject(bdevNvmeGetResp)
 }
+
+func BdevNvmeSetOptionsCmd() cli.Command {
+	return cli.Command{
+		Name:  "set-options",
```
Comment (shuo-wu) [inline app/cmd/basic/bdev_nvme.go:190]: NIT: `option-set`.

**Review by shuo-wu — APPROVED** [review]
LGTM

File: app/cmd/basic/bdev_nvme.go (hunk 2 above)
Comment (derekbit) [reply app/cmd/basic/bdev_nvme.go]: Updated. Thank you.

File: pkg/spdk/client/basic.go [hunk 3]
```diff
@@ -560,6 +560,35 @@ func (c *Client) BdevNvmeGetControllers(name string) (controllerInfoList []spdkt
 	return controllerInfoList, json.Unmarshal(cmdOutput, &controllerInfoList)
 }
 
+// BdevNvmeSetOptions sets global parameters for all bdev NVMe.
+// This RPC may only be called before SPDK subsystems have been initialized or any bdev NVMe
+// has been created.
+// Parameters, ctrlr_loss_timeout_sec, reconnect_delay_sec, and fast_io_fail_timeout_sec, are
+// for I/O error resiliency. They can be overridden if they are given by the RPC bdev_nvme_attach_controller.
+//
+// "ctrlrLossTimeoutSec": Controller loss timeout in seconds
```
Comment (innobead) [inline pkg/spdk/client/basic.go:569]: I assume the new options will only take effect after the controller reattached/restarted.

File: app/cmd/basic/bdev_nvme.go [hunk 4]
```diff
@@ -183,3 +184,51 @@ func bdevNvmeGet(c *cli.Context) error {
...
+			},
+			cli.IntFlag{
+				Name:  "fast-io-fail-timeout-sec",
+				Usage: "NVMe-oF controller fast I/O fail timeout in seconds for error cases",
+				Value: types.DefaultFastIOFailTimeoutSec,
+			},
+			cli.IntFlag{
+				Name:  "transport-ack-timeout",
+				Usage: "Time to wait ack until retransmission for RDMA or connection close for TCP. Range 0-31 where 0 means use default.",
+				Value: types.DefaultFastIOFailTimeoutSec,
```
Comment (innobead) [inline app/cmd/basic/bdev_nvme.go:211]: `DefaultTransportAckTimeout` instead?

**Review by innobead — CHANGES_REQUESTED** [review]
Just one comment needed to resolve.

File: pkg/spdk/client/basic.go (hunk 3 above)
Comment (derekbit) [reply pkg/spdk/client/basic.go:569]: Yes, you're right.

**Review by innobead — APPROVED** [review]

**Pull request merged**